File: examples/sample.cs
/**
 * Sample C# entities
 * Use XML doc comments: <pk />, <unique />, <notNull />, <default>, <ref>
 * Or Entity Framework annotations: [Key], [Required], [ForeignKey]
 */

using System;
using System.ComponentModel.DataAnnotations;

public class User
{
    /// <pk />
    [Key]
    public int Id { get; set; }

    /// <unique />
    /// <notNull />
    [Required]
    public string Username { get; set; }

    /// <unique />
    /// <notNull />
    [Required]
    public string Email { get; set; }

    /// <notNull />
    [Required]
    public string PasswordHash { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}

public class Post
{
    /// <pk />
    [Key]
    public int Id { get; set; }

    /// <notNull />
    [Required]
    public string Title { get; set; }

    public string? Body { get; set; }

    /// <ref table="User" column="Id" type="many-to-one" />
    [Required]
    public int AuthorId { get; set; }

    /// <default>draft</default>
    public string Status { get; set; } = "draft"; // draft, published, archived

    public DateTime? PublishedAt { get; set; }
    public DateTime CreatedAt { get; set; }
}

public class Comment
{
    /// <pk />
    [Key]
    public int Id { get; set; }

    /// <ref table="Post" column="Id" type="many-to-one" />
    [Required]
    public int PostId { get; set; }

    /// <ref table="User" column="Id" type="many-to-one" />
    [Required]
    public int UserId { get; set; }

    /// <notNull />
    [Required]
    public string Content { get; set; }

    public DateTime CreatedAt { get; set; }
}

File: examples/sample.ts
/**
 * Sample TypeScript entities
 * Use JSDoc annotations: @pk, @unique, @notNull, @default, @ref
 */

export interface User {
  /** @pk */
  id: number

  /** @unique @notNull */
  username: string

  /** @unique @notNull */
  email: string

  /** @notNull */
  passwordHash: string

  createdAt: Date
  updatedAt?: Date
}

export interface Post {
  /** @pk */
  id: number

  /** @notNull */
  title: string

  body?: string

  /**
   * @notNull
   * @ref User.id many-to-one
   */
  authorId: number

  /** @default 'draft' */
  status: 'draft' | 'published' | 'archived'

  publishedAt?: Date
  createdAt: Date
}

export interface Comment {
  /** @pk */
  id: number

  /**
   * @notNull
   * @ref Post.id many-to-one
   */
  postId: number

  /**
   * @notNull
   * @ref User.id many-to-one
   */
  userId: number

  /** @notNull */
  content: string

  createdAt: Date
}
